File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            int compli= target-nums[i];

            if(mp.find(compli)!=mp.end()){
                return {mp[compli],i};
            }
            else{
                mp[nums[i]]=i;
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        i=0;
        j= len(numbers)-1
        ans=[]
        while(i<=j):
            sum= numbers[i]+numbers[j]
            if(sum==target):
                ans.extend([i + 1, j + 1])
                break
                
            if(sum>target):
                j-=1

            else:
                i+=1

        return ans
            
                
        